### PR TITLE
Upgrades to cats 1.0.0-RC1

### PR DESCRIPTION
**Pull request by juanpedromoreno**
It also releases a new patch version.

### CODE REVIEW

**Comment by fedefernandez** [comment]
I think this is ready for review.

I have:
* Upgraded `catbird-util` and `finch-core`
* Updated the travis and sbt files with the latest Scala versions
* Changed the Monix documentation. The instance `Task.nondeterminism` has been removed from Monix. I've added a snippet about how to do parallel processing, but I'm not sure if we should remove the entire section.

@juanpedromoreno @peterneyens @raulraja

File: modules/core/jvm/src/test/scala/freestyle/nondeterminism.scala [hunk 1]
```diff
@@ -49,8 +49,9 @@ class NonDeterminismTests extends FunSuite with Discipline with Matchers {
       }
     }
 
-  implicit val throwableEq: Eq[Throwable] =
-    Eq[String].on(_.toString)
```
Comment (peterneyens) [inline modules/core/jvm/src/test/scala/freestyle/nondeterminism.scala:53]: I think this can be `Eq[String].by(_.toString)`.

File: build.sbt [hunk 2]
```diff
@@ -17,7 +17,7 @@ lazy val core = module("core")
   .crossDepSettings(
     commonDeps ++ Seq(
       %("cats-free"),
-      %("iota-core"),
+      %("iota-core", "0.3.3"),
```
Comment (raulraja) [inline build.sbt:20]: Are this behind in org-policies? shouldn't this changes happen there?

File: modules/tests/src/test/scala/MonixParallelTests.scala [hunk 3]
```diff
@@ -18,16 +18,14 @@ package freestyle.tests
...
-import monix.eval.Task.nondeterminism
 import monix.execution.Scheduler.Implicits.global
 import org.scalatest.{Matchers, WordSpec}
 import scala.concurrent.Await
 import scala.concurrent.duration.Duration
 
 class MonixParallelTests extends WordSpec with Matchers {
   "Applicative Parallel Support" should {
-    "allow non deterministic execution when interpreting to monix.eval.Task" in {
+    "allow non deterministic execution when interpreting to monix.eval.Task" ignore {
```
Comment (raulraja) [inline modules/tests/src/test/scala/MonixParallelTests.scala:28]: If this tests are not longer valid we should remove them.

**Review by raulraja — APPROVED** [review]

File: build.sbt (hunk 2 above)
Comment (fedefernandez) [reply build.sbt:20]: I've checked it, we can bump the sbt-freestyle plugin to `0.3.11` (we're currently using the `0.3.7` version) but the `0.3.11` uses the `0.3.15` version of `sbt-org-policies` where the selected `iota` version is `0.3.2`.

Two choices:
* Bump `sbt-freestyle` to `0.3.11`
* Release a new version of `sbt-freestyle` and use the new version here

File: modules/tests/src/test/scala/MonixParallelTests.scala (hunk 3 above)
Comment (fedefernandez) [reply modules/tests/src/test/scala/MonixParallelTests.scala:28]: The module is not aggregated in the `root` module, should I remove it?

File: build.sbt (hunk 2 above)
Comment (juanpedromoreno) [reply build.sbt:20]: IMHO, let's go for the faster path and let's do the `sbt-org-policies` upgrade after this is out. Makes sense?

**Comment by fedefernandez** [comment]
@peterneyens do you think we can move it forward?

**Comment by peterneyens** [comment]
We have to look a bit closer if we can leverage `Parallel` somehow (like mentioned in #463), but I think we can move forward for now.

**Comment by fedefernandez** [comment]
Sounds good, this PR won't close #463

**Pull request merged**